lib/ipam/ipam.go

```diff
@@ -672,20 +721,22 @@ func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, a
 	// If we need to support non-strict affinity and no auto-allocation of
 	// blocks, then we should query the actual allocation blocks and assign
 	// from those.
-	rem := num - len(ips)
+	rem := num - len(ia.IPs)
 	if config.StrictAffinity != true && rem != 0 {
```

Comment:
We could check the opposite of this condition and add a log stating that strict affinity is enabled but we were unable to find addresses in any of the affine blocks

Reply:
Basically move the "no free affine blocks" addMsg() call to here? I like that, think this is more clear. Will do, thanks!